tests/unit/APNSRequestMetadataTest.php

```diff
@@ -79,4 +79,124 @@ public function testThatUuidSetterWorks() {
 		$meta = $this->new_metadata()->setUuid( $uuid );
 		$this->assertEquals( $uuid, $meta->getUuid() );
 	}
+
+	public function testThatMetadataSerializationIncludesTopic() {
+		$topic = $this->random_string();
+		$encoded = $this->from_json( $this->new_metadata( $topic )->toJSON() );
+		$this->assertEquals( $topic, $encoded->topic );
+	}
+
+	public function testThatMetadataSerializationIncludesUuid() {
+		$uuid = $this->random_string();
+		$encoded = $this->from_json( $this->new_metadata( $this->random_string(), $uuid )->toJSON() );
+		$this->assertEquals( $uuid, $encoded->uuid );
+	}
+
+	public function testThatMetadataSerializationDoesNotIncludePushTypeByDefault() {
+		$encoded = $this->from_json( $this->new_metadata()->toJSON() );
+		$this->assertFalse( property_exists( $encoded, 'push_type' ) );
+	}
+
+	public function testThatMetadataSerializationIncludesPushTypeIfSpecified() {
+		$push_type = APNSPushType::MDM;
+		$encoded = $this->from_json( $this->new_metadata()->setPushType( $push_type )->toJSON() );
+		$this->assertEquals( $push_type, $encoded->push_type );
+	}
+
+	public function testThatMetadataSerializationDoesNotIncludeExpirationTimestampByDefault() {
+		$encoded = $this->from_json( $this->new_metadata()->toJSON() );
+		$this->assertFalse( property_exists( $encoded, 'expiration_timestamp' ) );
+	}
+
+	public function testThatMetadataSerializationIncludesExpirationTimestampIfSpecified() {
+		$expires_at = random_int( 1, time() );
+		$encoded = $this->from_json( $this->new_metadata()->setExpirationTimestamp( $expires_at )->toJSON() );
+		$this->assertEquals( $expires_at, $encoded->expiration_timestamp );
+	}
+
+	public function testThatMetadataSerializationDoesNotIncludePriorityByDefault() {
+		$encoded = $this->from_json( $this->new_metadata()->toJSON() );
+		$this->assertFalse( property_exists( $encoded, 'priority' ) );
+	}
+
+	public function testThatMetadataSerializationIncludesPriorityIfSpecified() {
+		$encoded = $this->from_json( $this->new_metadata()->setLowPriority()->toJSON() );
+		$this->assertEquals( APNSPriority::THROTTLED, $encoded->priority );
+	}
+
+	public function testThatMetadataSerializationDoesNotIncludeCollapseIdentifierByDefault() {
+		$encoded = $this->from_json( $this->new_metadata()->toJSON() );
+		$this->assertFalse( property_exists( $encoded, 'collapse_identifier' ) );
+	}
+
+	public function testThatMetadataSerializationIncludesCollapseIdentifierIfSpecified() {
+		$identifier = $this->random_string();
+		$encoded = $this->from_json( $this->new_metadata()->setCollapseIdentifier( $identifier )->toJSON() );
+		$this->assertEquals( $identifier, $encoded->collapse_identifier );
+	}
+
+	public function testThatMetadataDeserializationIncludesTopic() {
+		$topic = $this->random_string();
+		$meta = APNSRequestMetadata::fromJSON( $this->new_metadata( $topic )->toJSON() );
+		$this->assertEquals( $topic, $meta->getTopic() );
+	}
```

Comment:
It took me a bit to realize we moved from the serialization to the deserialization scope.

Would it make sense in the context of PHPUnit tests to split the test into one class for serialization and one for deserialization?

Reply:
No reason not to – it's really just a style thing. I personally find it a little _more_ confusing to split them up – I find I go looking for the test for a given file and can't find it in it's "companion" test file – it's like..oh yeah, it's in that other one.

My preferred approach here would be to move the serialization / deserialization into a separate class and then test that in isolation in a separate test.

Comments might also be a good solution here 😅 

Anyway, if this becomes problematic, let's split them up for sure, but I typically wouldn't reach for that straight away.

Reply:
> Anyway, if this becomes problematic, let's split them up for sure, but I typically wouldn't reach for that straight away.

Yeah. No problems at all. They can stay together.

> My preferred approach here would be to move the serialization / deserialization into a separate class and then test that in isolation in a separate test.

Two birds with one stone 😄 .